src/main/java/org/jenkinsci/plugins/pubsub/JobChannelMessage.java

```diff
@@ -80,20 +80,19 @@ public String getJobName() {
      * Jenkins {@link Item}.
      */
     public synchronized @CheckForNull Item getJobChannelItem() {
-        if (jobLookupComplete || jobChannelItem != null) {
-            return jobChannelItem;
+        if (jobLookupComplete || jobChannelItemFullName != null) {
+            return Jenkins.get().getItemByFullName(jobChannelItemFullName);
```

Comment:
Note that this is a behavior change: if the item was deleted, this method will begin returning null rather than returning an invalid object. In neither case would it fall through to the next clauses.

Reply:
The method could return a null before, not sure if it may have any actual implication